mempool/src/interface/mempool_interface.rs

```diff
@@ -72,6 +72,12 @@ pub trait MempoolInterface: Send + Sync {
     /// Get the fee rate such that it would put the new transaction in the top X MB of the mempool
     /// making it less likely to get rejected or trimmed in the case the mempool is full
     fn get_fee_rate(&self, in_top_x_mb: usize) -> Result<FeeRate, Error>;
+
+    /// Handle a p2p event.
+    // TODO This is here because of awkward inter-crate dependencies. A nicer solution would be to
+    // factor out subsystem interface traits into separate crates and subscribe to p2p events
+    // together with the other mempool setup code
+    fn handle_p2p_event(&mut self, evt: p2p_types::p2p_event::P2pEvent);
```

Comment:
Not very happy about exposing this via a public interface. See the PR description for details.